### PR TITLE
Included MassAssignmentSecurity in mixins if ::ProtectedAttributes is defined

### PR DESCRIPTION
**Pull request by knovoselic**
Fixed problem described in #541.
This is alternative approach to #555 .

### CODE REVIEW

**Comment by knovoselic** [comment]
@tute What do you think about this one?

**Comment by tute** [comment]
This looks ok to me. An alternative I like better is the one I quote from https://github.com/doorkeeper-gem/doorkeeper/pull/555#issuecomment-70371765:

> What I would love is not need parameters protection in the codebase, remove this lines, and anywhere we call `save` or the `update` flavors, inline each attribute to make sure this vulnerability doesn't exist.

Will think of this in two weeks, when I assign some workweek time for doorkeeper. Thanks for your input!

**Comment by tute** [comment]
I think the condition we want here is that `ProtectedAttributes` is defined, and also the module is not included yet. Otherwise we'll include it twice in ActiveRecord.

**Comment by tute** [comment]
This looks good to merge. Three steps to do before:

1. Can you please improve your commit message, with context on the three related issues/PRs, before?
2. Can you update the CHANGELOG?
3. Can you rebase on top of latest master?

Thank you!

**Comment by knovoselic** [comment]
Hey, unfortunately I don't have enough time to work on it, should be able in two weeks or so. I'll let you know when it's done. Thanks.

**Comment by tute** [comment]
ping @knovoselic. Thanks again for your good contribution!

**Comment by tute** [comment]
@knovoselic after your change, we can assume in https://github.com/doorkeeper-gem/doorkeeper/blob/master/lib/doorkeeper/models/access_grant_mixin.rb#L14 that if it responds to `attr_accessible` it can safely be called, simplifying the condition. Would you update those, please?

**Comment by tute** [comment]
ping @knovoselic. Thank you!

**Comment by knovoselic** [comment]
@tute Sorry for prolonging this so much, I'll do it 100% next weekend.

**Comment by tute** [comment]
> I'll do it 100% next weekend.

That's this weekend! :stuck_out_tongue:

**Comment by knovoselic** [comment]
Yep, I'll work on it today :) Thanks for being patient.

**Comment by knovoselic** [comment]
@tute Hey, I think I've done everything you've asked me to do. Any idea why this build has failed?
https://travis-ci.org/doorkeeper-gem/doorkeeper/builds/61950209
The next one has passed, and I haven't updated anything related to the code that has failed...

**Comment by tute** [comment]
Second commit ("Simplified condition...") shouldn't be there, as it fails. See https://github.com/doorkeeper-gem/doorkeeper/pull/555#issuecomment-70370978. After you remove that, I'll merge. Thank you! :smiley:

**Comment by tute** [comment]
Regarding the spec, it's a flappy test, I'm working on it in the "extract ORMs" PR. Thank you!

File: NEWS.md [hunk 1]
```diff
@@ -4,6 +4,11 @@ User-visible changes worth mentioning.
 
 ---
 
+## (unreleased)
+
+- [#541] Fixed `undefined method attr_accessible` problem on Rails 4
```
Comment (tute) [inline NEWS.md]: Let's add #599 too to the NEWS.

**Comment by knovoselic** [comment]
@tute  I've added #599 to the news. I'm a little confused by this message:
> Second commit ("Simplified condition...") shouldn't be there, as it fails. See #555 (comment). After you remove that, I'll merge. Thank you! :smiley:

I do not see it failing, so I'm not sure if there's something that's not covered by the tests? If there is, what were you thinking about when you've suggested that those lines can be simplified?

**Comment by tute** [comment]
> I do not see it failing, so I'm not sure if there's something that's not covered by the tests? If there is, what were you thinking about when you've suggested that those lines can be simplified?

You are totally right, we don't care any longer because we don't test against end-of-life'd Rails 4 (where that situation happened).
Thanks!

**Comment by knovoselic** [comment]
Thanks :) When do you think it will be released?

**Comment by tute** [comment]
Past week I released 2.2.1; I'd wait at least two weeks and two patches before releasing next version. Meanwhile you can point your Gemfile to this specific merge commit. Thanks!

**Comment by knovoselic** [comment]
Will do, thanks!

**Pull request merged**